backend/src/main/kotlin/com/tngtech/apicenter/backend/config/RestResponseExceptionHandler.kt

```diff
@@ -1,53 +1,48 @@
 package com.tngtech.apicenter.backend.config
 
+import com.tngtech.apicenter.backend.domain.exceptions.SpecificationNotFoundException
+import com.tngtech.apicenter.backend.domain.exceptions.SpecificationParseException
+import com.tngtech.apicenter.backend.domain.exceptions.VersionAlreadyExistsException
 import org.springframework.http.HttpStatus
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.ControllerAdvice
 import org.springframework.web.bind.annotation.ExceptionHandler
 import java.time.Instant
 import java.time.format.DateTimeFormatter
 import mu.KotlinLogging
-import java.util.*
 
 private val logger = KotlinLogging.logger {  }
-class SpecificationNotFoundException(val specificationId: UUID, val version: String? = "") : RuntimeException()
-class SpecificationParseException(val msg: String) : RuntimeException()
-class VersionAlreadyExistsException(val specificationTitle: String) : RuntimeException()
 
 data class ErrorMessage(
-        val message: String,
+        val httpReasonPhrase: String,
+        val userMessage: String,
         val timestamp: String = DateTimeFormatter.ISO_INSTANT.format(Instant.now())
 )
 
+fun responseBuilder(userMessage: String, status: HttpStatus): ResponseEntity<ErrorMessage> {
+    return ResponseEntity(
+        ErrorMessage(status.reasonPhrase, userMessage),
+        null,
+        status
+    )
+}
+
 @ControllerAdvice
 class RestResponseExceptionHandler {
 
     @ExceptionHandler(SpecificationNotFoundException::class)
     fun handleNotFound(exc: SpecificationNotFoundException): ResponseEntity<ErrorMessage> {
-        logger.info("Specification ${exc.specificationId} ${exc.version} not found")
-        return ResponseEntity(
-                ErrorMessage("Specification not found"),
-                null,
-                HttpStatus.NOT_FOUND
-        )
+        val detailedMessage = "Specification ${exc.specificationId} ${exc.version} not found"
+        logger.info(detailedMessage, exc)
```

Comment:
If you inline the string variable, it should be a bit more performant. Also, I think an `error` is more appriopriate :)